[wasm] Update Wasm.Build.Tests to build with tfm=net9.0

radical opened the pull request:
Currently the test projects in Wasm.Build.Tests are built with `tfm=net8.0`, since that's what the sdk supported. But to run the tests against the `artifacts` from the runtime build, the various workload packs were replaced by a couple of workarounds.

- Here we update all the non-blazor projects to now build with `tfm=net9.0`. And disable the corresponding workaround in the workload by setting `ForceNet8Current=false` in WBT.
- The same is not done for blazor projects because they don't support building with `net9.0` yet. And thus the workaround to enable testing them with `artifacts` - `ForceNet8Current=true` is unchanged.

## Details of changes

-  workload-testing.targets: Make dotnet script paths available outside
    targets. This is useful when `dotnet-none` is already installed and
    thus the installation target does not run.
- [wasm] Improve parsing runtime pack versions from environment variables
- [wasm] WBT: Remove --experiment-wasm-simd for v8
- [wasm] Fix wasmbrowser template so it correctly updates the tfm in csproj
- [wasm] WBT: Share nuget config files between wasm and wasi

Fixes https://github.com/dotnet/runtime/issues/91357 .

lewing commented on src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.targets.in:
```diff
@@ -8,7 +8,7 @@
     <TargetsCurrent Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionEquals('$(TargetFrameworkVersion)', '9.0'))">true</TargetsCurrent>
     <!-- force net8 to use the current targets until the SDK can target net9 -->
-    <ForceNet8Current Condition="'$(ForceNet8Current)' == ''">true</ForceNet8Current>
+    <ForceNet8Current Condition="'$(ForceNet8Current)' == ''">false</ForceNet8Current>
```
we may just want to explicitly set this to false in wbt until sdk->installer is flowing but this is fine while you are working on this pr

radical replied:
I'll revert this change, and move the setting to wbt.

radical commented:
/azp run runtime-wasm-perf

radical committed:
workload-testing.targets: Make dotnet script paths available outside

.. targets. This is useful when `dotnet-none` is already installed and
thus the installation target does not run.

radical committed:
WorkloadManifest.targets: Disable workaround to use net9.0 artifacts

.. with `tfm=net8.0` . But don't disable this for Blazor projects which
are still using `tfm=net8.0`.

radical committed:
[wasm] Improve parsing runtime pack versions from environment variables

radical committed:
[wasm] Allow WBTOverridePack targets use to be set per test

This allows having it enabled only for the blazor projects right now,
which are still on tfm=net8.0 .

radical committed:
[wasm] WBT: Update tests to run with net9.0

radical committed:
[wasm] WBT: Remove --experiment-wasm-simd for v8

radical committed:
[wasm] WBT: differentiate version used for tasks from the tfm

radical committed:
[wasm] Fix wasmbrowser template so it correctly updates the tfm in csproj

radical committed:
[wasm] WBT: Add runtime pack version for 9 in wasi

radical committed:
[wasm] WBT: Share nuget config files between wasm and wasi

radical committed:
[wasm] WBT: Fix blazor tests

radical committed:
[wasm] WBT: Don't test 8.0 templates yet. this will need more changes

pavelsavara commented:
please provide some description of the changes in this PR

pavelsavara commented on src/mono/wasm/templates/templates/browser/browser.0.csproj:
```diff
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk.WebAssembly">
   <PropertyGroup>
-    <TargetFramework>net8.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
```
could you please elaborate ?

radical replied:
https://github.com/dotnet/runtime/blob/77d836c65565c389b0533acbf340635d83ae6ab9/src/mono/wasm/templates/templates/browser/.template.config/template.json#L48-L51

This is what we have in the template config, and it expects to replace `"net7.0"` string. It was incorrectly changed to `net8.0`, but it didn't affect anything because the replacement value was also `net8.0`.

radical committed:
[wasm] Don't default ForceNet8Current=false yet

Instead set this only for WBT.

radical committed:
perf: add dummy change to trigger runtime-wasm-perf
